batch/batch/front_end/front_end.py

```diff
@@ -1503,6 +1526,27 @@ async def delete_batch_loop_body(app):
     return should_wait
 
 
+async def monitor_billing_limits_loop_body(app):
+    db = app['db']
+    while True:
+        records = await _query_billing_limits(app)
+        for record in records:
+            limit = record['limit']
+            accrued_cost = record['accrued_cost']
+            if limit is not None and accrued_cost >= limit:
+                running_batches = db.execute_and_fetchall('''
+SELECT id, user
+FROM batches
+WHERE billing_project = %s AND state = 'running';
+''',
+                                                          (record['billing_project'],))
+                async for batch in running_batches:
+                    await _cancel_batch(app, batch['id'], batch['user'])
+
+        # what should the time be here? There's on average 3 batch front-end instances.
+        await asyncio.sleep(30)
```

Comment:
This kind of feels like something the driver should do. Sometimes we scale up to ~10 front ends. We don't want to hammer the DB with 10 of these queries.